### PR TITLE
Implementation of the Google OAuth on the frontend 

### LINKED ISSUE #77: Implementation of the Google OAuth on the frontend
**Describe the bug**
Although the user login view has been implemented in #5, the Google OAuth button and authentication process have not been implemented.

**To Reproduce**
1. Go to home
2. Click on "Become a mentor" button
3. Scroll down to bottom

**Expected behavior**
Need you display that Google Sign in button in this Login modal

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Are you working on it?**
Yes

### PR DESCRIPTION
**Pull request by mayura-andrew**
## Purpose

The purpose of this PR is to fix #77
## Goals
Implemented Google Sign-in button in the LoginModal Component to enhance user experience and ease of login.

## Approach
Integrated Google's OAuth API to handle authentication and incorporated the Google Sign-In button into the LoginModal component. Users can now sign in using their Google accounts.

## Screenshots
Screenshot_20240208_222432

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Test environment
Web Browser - Vivaldi 6.5.3206.59 stable
NodeJs - v20.10.0

## Learning
Learned about Google's OAuth implementation in frontend. This feature will help users to login quickly using their Google account credentials, improving the overall user experience.

### CODE REVIEW

File: src/components/LoginModal/index.tsx [hunk 1]
```diff
@@ -140,6 +145,31 @@ const LoginModal: React.FC<LoginModalProps> = ({ handleClose }) => {
                 >
                   Login
                 </button>
+                <div
+                  className="my-1 flex items-center before:mt-0.2 before:flex-1 before:border-t before:border-neutral-300 after:mt-0.5 after:flex-1 after:border-t after:border-neutral-300"
+                  style={{ margin: '0' }}
```
Comment (anjula-sack) [inline src/components/LoginModal/index.tsx]:
```suggestion
```
m-0

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by anjula-sack** [comment]
Hi @mayura-andrew, can you add the requested change?

**Comment by mayura-andrew** [comment]
Year sure I will it tonight

**Review by anjula-sack — APPROVED** [review]
Good job! @mayura-andrew

**Pull request merged**